test/ecto/migrator_test.exs

```diff
@@ -431,4 +465,35 @@ defmodule Ecto.MigratorTest do
       assert run(TestRepo, [{13, ChangeMigration}, {14, UpDownMigration}], :up, to: 15, log: false) == [13, 14]
     end
   end
+
+  describe "migration callbacks" do
+    test "both run when in a transaction going up" do
+      log = capture_log(fn ->
+        assert up(TestRepo, 10, MigrationWithCallbacks) == :ok
+      end)
+
+      assert log =~ "after_begin"
+      assert log =~ "before_commit"
+    end
+
+    test "they are both run in a transaction going down" do
+      assert up(TestRepo, 10, MigrationWithCallbacks, log: false) == :ok
+
+      log = capture_log(fn ->
+        assert down(TestRepo, 10, MigrationWithCallbacks) == :ok
+      end)
+
+      assert log =~ "after_begin_down"
+      assert log =~ "before_commit_down"
+    end
+
+    test "they are not run when the migration is disabled" do
```

Comment:
I would remove the "they" from the title as otherwise the description would be "migration callbacks they are not run ...".